### PR TITLE
Fix hidden state size for StaticWhisperPipeline

### PR DESCRIPTION
**Pull request by eshiryae**

### CODE REVIEW

File: src/cpp/src/whisper_pipeline_static.cpp [hunk 1]
```diff
@@ -394,7 +394,17 @@ void add_attention_mask_input(std::shared_ptr<ov::Model> model) {
     pm.run_passes(model);
 }
 
-void reshape_to_static(std::shared_ptr<ov::Model> model, const uint32_t input_size, const uint32_t kvcache_size) {
+
+uint32_t get_encoder_hidden_state_size(const std::shared_ptr<ov::Model>& encoder) {
```
Comment (TolyaTalamanov) [inline src/cpp/src/whisper_pipeline_static.cpp:398]: Simply
```
encoder->output("last_hidden_state");
```

File: src/cpp/src/whisper_pipeline_static.cpp [hunk 2]
```diff
@@ -394,7 +394,17 @@ void add_attention_mask_input(std::shared_ptr<ov::Model> model) {
...
+    for (auto output : encoder->outputs()) {
+        if (output.get_any_name() == "last_hidden_state") {
+            return output.get_partial_shape()[1].get_length();
```
Comment (soumendukrg) [inline src/cpp/src/whisper_pipeline_static.cpp:401]: Need to get both dimension 1 and dimension 2 of encoder output. Dim 1 will be 1500. Dim 2 will vary based on model version. For example, dim 2 = 512 for base, 768 for small.

File: src/cpp/src/whisper_pipeline_static.cpp [hunk 3]
```diff
@@ -411,14 +421,14 @@ void reshape_to_static(std::shared_ptr<ov::Model> model, const uint32_t input_si
             const auto& partial_shape = input.get_partial_shape();
             new_shape = partial_shape;
             new_shape[0] = 1;     // batch_dim
-            new_shape[1] = 1500;  // FIXME: is it got from encoder output{'last_hidden_state'}
+            new_shape[1] = lhstate_size;  // from encoder output{'last_hidden_state'}
```
Comment (soumendukrg) [inline src/cpp/src/whisper_pipeline_static.cpp:424]: Need to assign new_shape[2] = last_hidden_state.shape[2]

**Review by soumendukrg — CHANGES_REQUESTED** [review]
Need the additional changes needed.

**Comment by RyanMetcalfeInt8** [comment]
Can we try to get this fix pulled into releases/2024/5 branch as well?

**Review by KingRei — COMMENTED** [review]
[whisper_pipeline_static.zip](https://github.com/user-attachments/files/17698557/whisper_pipeline_static.zip)
I added our modification to avoid converting dynamic shape to static shape failure in following reference code.

**Comment by dmatveev** [comment]
@RyanMetcalfeInt8  propagated to release branch - #1186

**Review by dmatveev — APPROVED** [review]
LGTM

**Review by TolyaTalamanov — APPROVED** [review]

**Review by soumendukrg — APPROVED** [review]
The modifications look good. Thanks.
